docs/source/virtual_rainforest/core/data.md

```diff
@@ -0,0 +1,153 @@
+loading and storage functions for the Virtual Rainforest. The data system is extendable
+to provide support for different file formats and axis validation (see the [module API
+docs](../../api/core/data.md)) but that is beyond the scope of this document.
+
+A Virtual Rainforest simulation will have one instance of the
+{class}`~virtual_rainforest.core.data.Data` class and this instance behaves as a
+dictionary to provide access to the different forcing and internal variables used in the
+simulation. All of the variables are stored as {class}`~xarray.DataArray` objects from
+the {mod}`xarray` package, which provides a consistent indexing and data manipulation
+for the underlying arrays of data.
+
+In many cases, a user will simply provide a Data configuration file to set up the data
+that will be validated and loaded when a simulation runs, but the main functionality for
+working with data using Python are shown below.
+
+## Core data axes
+
+One of the main functions of the {mod}`~virtual_rainforest.core.data` module is to make
+sure that any data source is congruent with the basic configuration of a simulation:
+things like the size and layout of the spatial grid or the the duration and spacing of
+temporal data. The {mod}`~virtual_rainforest.core.data` module maintains a registry of
+these core axes along with a library of validators that can be used to map a DataArray
+onto the core axes. These validators are applied automatically when a DataArray is
+loaded, using the dimension names and coordinates of the DataArray object to identify an
+appropriate validator.
+
+## Structure of `DataArray` objects
+
+A {class}`~xarray.DataArray` provides a single variable as an array: each dimension of
+the array must have at least a dimension name and can also have coordinates along the
+dimension. The three examples below show how data arrays can be created with (and
+without) dimension names and coordinates and shows how the variable name for the array
+is set.
+
+```{code-cell}
+from xarray import DataArray
+import numpy as np
+
+# No dimension names specified, so default names are assigned and no variable name
+d = DataArray(np.ones((5, 5)))
+print(d)
+```
+
+```{code-cell}
+# A data array with just dimension names
+d = DataArray(np.ones((5, 5)), dims=("y", "x"))
+print(d)
+```
+
+```{code-cell}
+# A _named_ array with coordinates along both axes
+d = DataArray(
+    np.ones((5, 5)),
+    name="temperature",
+    coords={"y": np.arange(0, 401, 100), "x": np.arange(0, 401, 100)},
+)
+print(d)
+```
+
+## Creating a `Data` instance
+
+A  {class}`~core.data.Data` instance is created using information that provides
+information on the core configuration of the simulation. At present, this is just the
+spatial grid being used.
+
+```{code-cell}
+from virtual_rainforest.core.grid import Grid
+from virtual_rainforest.core.data import Data
+
+# Create a simple default grid and a Data instance
+grid = Grid()
+data = Data(grid=grid)
+
+print(data)
+```
```

Comment:
It might be worth noting that probably most of the input data will not be in the correct resolution and there needs to be a mapping step between loading data and running the model. Alternatively, it should be mentioned that this is not done in the package but with a preprocessing step.

Reply:
I think there need to be two steps: recipes for preprocessing, but also validation that the data has been preprocessed to align with key axes correctly.